src/token.rs

```diff
@@ -810,6 +810,32 @@ pub fn get_next_token(
     result
 }
 
+/// Test if the given character is a hex character.
+fn is_hex_char(c: char) -> bool {
+    match c {
+        'a'..='f' => true,
+        'A'..='F' => true,
+        '0'..='9' => true,
+        _ => false,
+    }
+}
+
+/// Test if the given character is an octal character.
+fn is_octal_char(c: char) -> bool {
+    match c {
+        '0' | '1' => true,
```

Comment:
Probably needs `'0'..'7'`?

Reply:
You are of course correct. I'll try and figure out why the test doesn't break before fixing it. Thanks!

Reply:
Ok, so the reason it parsed is because when an escape sequence is encountered, it loops around and continues consuming numerical literals.

This actually makes me think that lexing of numericals might benefit from being simplified further: simply identify the kind of numeric literal and consumer everything alphanumerical until whitespace.

This appears to be how Rust does it: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c019d6741c5c4b6167bcc352be480ad1

For now I just fixed the immediate issue, thanks!

Reply:
Not very comforting... because it should `break` out of the loop on the first invalid character...

But still, doesn't see anything wrong as yet, so merging this!